### DISCUSSION TITLE
Check Spelling in Starlight Docs

### DISCUSSION THREAD

**Discussion by trueberryless** [withastro/starlight#1068] [Feature Requests]
Over the last week I implemented the Github Action [spell-ckecking](https://github.com/check-spelling/check-spelling) into my [own repo](https://github.com/trueberryless-org/mutanuq) which uses Starlight.

I think it would be great to also implement such a feature in the Starlight Docs and I am willing to create a PR which implements the check-spelling Action if the community decides that Starlight should implement this feature.

**Comment by at-the-vr** [comment]
I am neutral on this case, its good to have but with all the linting and formatting I find myself delayed overtime. I find this in my lowest priorities (subjectively). It is nicely maintained and documented 💯

**Comment by delucis** [comment]
This is a cool idea! I wonder if you could publish this as a community add-on package a bit like https://github.com/HiDeoo/starlight-links-validator ?

**Reply by trueberryless** [reply]
That would also be worth thinking about.

However, the spell checker has many Github related features, like answering on a PR, showing comments in the files and commenting on pushes to main branch. I'm not sure if these benefits can also be applied to a Astro package...

**Reply by HiDeoo** [reply]
> However, the spell checker has many Github related features, like answering on a PR, showing comments in the files and commenting on pushes to main branch.

Maybe this could be bundled and published as a GitHub Action available on the GitHub Marketplace?

**Reply by trueberryless** [reply]
That's the wonderful thing about it: It already is.

It is called check-spelling action and here is the link: https://github.com/check-spelling/check-spelling
At least that's what I used for my repo...

**Reply by HiDeoo** [reply]
Oh I should have been more clear. I meant either a JS or composite GitHub Action that would re-use this one but with extra features dedicated to Starlight, e.g. automatic configuration of the dictionnaries based on the Starlight configured locales, etc.

**Reply by trueberryless** [reply]
I can give you some examples of my repo where this action helps out a lot:

- Commenting on commits: https://github.com/trueberryless-org/mutanuq/commit/dbe0c2bd32780ac8b81b43c187086fcb03256109#commitcomment-132435149
- Commenting on Pull Request (with updates when new commits): https://github.com/trueberryless-org/mutanuq/pull/35#issuecomment-1811902076
- Creating entries in Security tab for misspellings: https://github.com/trueberryless-org/mutanuq/security/code-scanning?query=is%3Aclosed+branch%3Amain

and much more. You can see **every feature** of this Github Action right [here](https://docs.check-spelling.dev/Showcase.html)

**Reply by trueberryless** [reply]
> Oh I should have been more clear. I meant either a JS or composite GitHub Action that would re-use this one but with extra features dedicated to Starlight, e.g. automatic configuration of the dictionaries based on the Starlight configured locales, etc.

Okay, this is a very good idea! Haven't though about that one before...

**Reply by jasikpark** [reply]
> [HiDeoo/starlight-links-validator](https://github.com/HiDeoo/starlight-links-validator)

does this work for generic astro sites as well?

**Reply by HiDeoo** [reply]
> does this work for generic astro sites as well?

I did not try it but in the current state, I think it could (or at least should be able with not a lot of modifications) altho my plan is to convert the integration into a Starlight plugin when possible to add more Starlight specific feature like support for Starlight fallback pages and same-locale links (which would use the Starlight locales config). At this point, this would no longer an option at all.

**Reply by trueberryless** [reply]
> This is a cool idea! I wonder if you could publish this as a community add-on package a bit like https://github.com/HiDeoo/starlight-links-validator ?

Looking back at this reply today, thanking you so much for giving me this wonderful idea back then!
Done: https://starlight-spell-checker.trueberryless.org/

**Comment by trueberryless** [comment]
If you're interested, check out my Starlight plugin, which makes this possible:
https://starlight-spell-checker.trueberryless.org/

**Reply by delucis** [reply]
Amazing work 👏

**Reply by trueberryless** [reply]
Thank you!

I just noticed that it is REALLY SLOW rn, gonna have to do some optimisation sometimes...
